ResearchKit/Common/ORKFormStep.h

```diff
@@ -129,6 +129,14 @@ ORK_CLASS_AVAILABLE
 @property (nonatomic, copy, readonly) NSString *identifier;
 
 /**
+ A Boolean value indicating whether the form item is optional.
+ 
+ The default value of this property is `NO`. When the value is `YES`, this form item doesn't need
```

Comment:
I see your point. 

In this PR we need fill a matrix:

| Step         | FormItem           | Require to answer?  |
| ------------- |:-------------:| -----:|
| optional    | not-optional | ? |
| optional    | optional      |   NO  |
| not-optional    | not-optional     |  YES  |
| not-optional    | optional      |   ? |

And since we have the granularity control for each `formItem`, shall we ignore `optional` setting on step level in a form step? If so, default value `YES` make more sense.